agent/service_manager.go

```diff
@@ -326,7 +326,7 @@ func makeConfigRequest(bd BaseDeps, addReq AddServiceRequest) *structs.ServiceCo
 		// Also if we have any upstreams defined, add them to the request so we can
 		// learn about their configs.
 		for _, us := range ns.Proxy.Upstreams {
-			if us.DestinationType == "" || us.DestinationType == structs.UpstreamDestTypeService {
+			if us.DestinationType == "" || us.DestinationType == structs.UpstreamDestTypeService && !us.CentrallyConfigured {
```

Comment:
Could you please add some extra parentheses here? I usually find it helps me not have to go look up operator precedence rules every time to ensure I'm looking at an intended grouping instead of an accidental bug.

Reply:
This was resolved by not needing to exclude centrally configured upstreams since they don't get merged into the original NodeService